### PR TITLE
Changed history_count_mode ENUM setup

### PR DESCRIPTION
**Pull request by jaapen**
I have revised the branch and your implementation now works with the ENUMS as suggested in the pull request. Can you merge the pull request into your branch and then reopen the pull request https://github.com/hapifhir/hapi-fhir-jpaserver-starter/pull/500 again?

### CODE REVIEW

**Comment by jaapen** [comment]
@sabukar-fn what do you think about the implementation?

**Comment by SamatarAbukar** [comment]
Looks good to me!

**Comment by SamatarAbukar** [comment]
> I have revised the branch and your implementation now works with the ENUMS as suggested in the pull request. Can you merge the pull request into your branch and then reopen the pull request [hapifhir#500](https://github.com/hapifhir/hapi-fhir-jpaserver-starter/pull/500) again?

I no longer have access to the sabukar-fn github account, so you're better off just creating a new PR with this code.

File: src/main/resources/application.yaml [hunk 1]
```diff
@@ -132,10 +132,10 @@ hapi:
...
         - '*'
     # history_count_mode is an integer that represents the mode that the DAOConfig will use when getting the history count during a history request
-      # history_count_mode = 0 -> CACHED_ONLY_WITHOUT_OFFSET (default within hapi-fhir DAOConfig)
-      # history_count_mode = 1 -> COUNT_ACCURATE (ignores cache and always gets correct count)
-      # history_count_mode = 2 -> COUNT_DISABLED (does not return count at all)
-    history_count_mode: 1
+      # history_count_mode = CACHED_ONLY_WITHOUT_OFFSET (default within hapi-fhir DAOConfig)
+      # history_count_mode = COUNT_ACCURATE (ignores cache and always gets correct count)
+      # history_count_mode = COUNT_DISABLED (does not return count at all)
+    history_count_mode: CACHED_ONLY_WITHOUT_OFFSET
```
Comment (sabukar-fn) [inline src/main/resources/application.yaml]:
```suggestion
    history_count_mode: COUNT_ACCURATE
```

**Pull request merged**